kiwi/command.py

```diff
@@ -165,3 +173,24 @@ def _select():
             error_available=error_available(),
             process=process
         )
+
+    @classmethod
+    def _command_exists(self, command, env):
+        """
+        Tests if the given command is executable for the specified
+        environment. If the specified environment does not include
+        'PATH' variable, the method raises a KiwiCommandError
+        exception
+
+        :param string command: command to validate
+        :param list env: custom os.environ
+
+        :return: True if the command is found, False otherwise
+        :rtype: bool
+        """
+        if 'PATH' not in env:
+            raise KiwiCommandError('No PATH variable in environment')
+        return any(
+            os.access(os.path.join(path, command), os.X_OK)
+            for path in env['PATH'].split(os.pathsep)
+        )
```

Comment:
right, so I suggest to exend Path.which by a custom_env and a flags parameter. If custom_env is provided it is used instead of os.environ. Does that makes sense ?